### PR TITLE
Wagtail 4.2 support

### PR DESCRIPTION
**Pull request by katdom13**
Supress warnings from earlier wagtail versions

### CODE REVIEW

**Comment by Stormheg** [comment]
Hi @katdom13, thanks for the contribution! ❤️

I'll check this out in detail when I have time. Please be patient.

File: tox.ini [hunk 1]
```diff
@@ -3,20 +3,17 @@ linting_folders=src/wagtailbakery/ tests/ examples/
...
 deps=
-  django32: django>=3.2,<3.3       # WT 2.15 LTS
-  django40: django>=4.0,<4.1       # WT 4.0, WT 4.1 LTS
-  django41: django>=4.1,<4.2       # WT 4.0, WT 4.1 LTS
-  wagtail215: wagtail>=2.15,<2.16  # Old LTS
-  wagtail40: wagtail>=4.0,<4.1  # Current latest
+  django32: django>=3.2,<3.3
+  django40: django>=4.0,<4.1
+  django41: django>=4.1,<4.2
   wagtail41: wagtail>=4.1rc1,<4.2  # Current LTS
```
Comment (nickmoreton) [inline tox.ini:15]:
```suggestion
  wagtail41: wagtail>=4.1,<4.2  # Current LTS
```

File: tox.ini [hunk 2]
```diff
@@ -3,20 +3,17 @@ linting_folders=src/wagtailbakery/ tests/ examples/
...
+  wagtail42: wagtail>=4.2rc1,<4.3
```
Comment (nickmoreton) [inline tox.ini:16]:
```suggestion
  wagtail42: wagtail>=4.2,<5.0
```

**Review by nickmoreton — COMMENTED** [review]
@katdom13 Thanks the code changes look good to me. Just a couple of suggestions.

File: tox.ini (hunk 1 above)
Comment (katdom13) [reply tox.ini]: Thanks, this is now fixed in the latest commit https://github.com/wagtail/wagtail-bakery/pull/79/commits/6ce7b17599914c80a919729c3c10a738593b7b4a
as well as the one below.

**Review by nickmoreton — APPROVED** [review]
LGTM 👍

File: examples/aws/example/models.py [hunk 3]
```diff
@@ -1,7 +1,8 @@
-from wagtail.admin.edit_handlers import FieldPanel, StreamFieldPanel
-from wagtail.core import blocks
-from wagtail.core.fields import StreamField
-from wagtail.core.models import Page
+from wagtail import blocks
+from wagtail.admin.panels import FieldPanel
+from wagtail.admin.panels import FieldPanel as StreamFieldPanel
```
Comment (Stormheg) [inline examples/aws/example/models.py:3]: **Suggestion:** we can drop the StreamFieldPanel import and use FieldPanel in the example. I feel like this aliased import may confuse users reading this example.

File: examples/multisite/example/models.py [hunk 4]
```diff
@@ -1,7 +1,8 @@
-from wagtail.admin.edit_handlers import FieldPanel, StreamFieldPanel
-from wagtail.core import blocks
-from wagtail.core.fields import StreamField
-from wagtail.core.models import Page
+from wagtail import blocks
+from wagtail.admin.panels import FieldPanel
+from wagtail.admin.panels import FieldPanel as StreamFieldPanel
```
Comment (Stormheg) [inline examples/multisite/example/models.py:3]: See above comment.

File: examples/site/example/models.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
-from wagtail.admin.edit_handlers import FieldPanel, StreamFieldPanel
-from wagtail.core import blocks
-from wagtail.core.fields import StreamField
-from wagtail.core.models import Page
+from wagtail import blocks
+from wagtail.admin.panels import FieldPanel, StreamFieldPanel
```
Comment (Stormheg) [inline examples/site/example/models.py:2]: See above comment.

File: tests/conftest.py [hunk 6]
```diff
@@ -20,17 +19,12 @@ def pytest_configure():
...
     ]
-    if WAGTAIL_VERSION >= (2, 9):
-        wagtail_middleware = [
-            'wagtail.contrib.redirects.middleware.RedirectMiddleware',
-        ]
-    else:
-        wagtail_middleware = [
-            'wagtail.core.middleware.SiteMiddleware',
-            'wagtail.contrib.redirects.middleware.RedirectMiddleware',
-        ]
```
Comment (Stormheg) [inline tests/conftest.py:33]: Wow, that is old compatibility code. Thanks for cleaning that up! 🚀

**Review by Stormheg — COMMENTED** [review]
Hi @katdom13 - sorry that it took me so long to properly review your pull request.

I'm really happy with your changes, but I do have one minor suggestion (see comment). Are you happy to make that change?

Thanks again for your help maintaining wagtail-bakery ❤️

**Review by Stormheg — APPROVED** [review]

**Comment by loicteixeira** [comment]
Thank you for your work @katdom13 and the review @Stormheg.
I'll attempt to finish #80 and create a new release.

**Pull request merged**